docs/badges/configuration.rst

```diff
@@ -0,0 +1,232 @@
+        "collectors": [
+            "credentials.apps.badges.collecting.BadgeCollector",
+            "credentials.apps.badges.distribution.credly.CredlyBadgeCollector",
+        ],
+        "events": [
+            "org.openedx.learning.course.grade.passed.v1",
+            "org.openedx.learning.course.grade.failed.v1",
+        ],
+    }
+
+Configuration describes:
+
+- `processor`_ - implementation of incoming events processing pipeline.
+- `collector`_ - implementation of badging progress collecting and side effects.
+- ``events`` - explicit event types subset of *incoming* events.
+
+Incoming Events configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+1. Listed incoming event types become available for requirements configuration.
+2. Listed incoming event types are taken into account during processing - `Badge Processor`_ is auto-connected to listen to those event types.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["events"] = [
+        "org.openedx.learning.course.grade.passed.v1",
+        "org.openedx.learning.course.grade.failed.v1",
+    ]
+
+
+Outgoing Events configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Optionally, Badges configuration lists event types which must be registered for external emission.
+
+.. code-block:: sh
+
+    EVENT_BUS_PRODUCER_CONFIG = {
+        ...
+
+        "org.openedx.learning.badge.awarded.v1": {
+            "badges": {"event_key_field": "badge.uuid", "enabled": True },
+        },
+        "org.openedx.learning.badge.revoked.v1": {
+            "badges": {"event_key_field": "badge.uuid", "enabled": True },
+        },
+    }
+
+To propagate out of Credentials service `EVENT_BUS_PRODUCER_CONFIG` must be extended.
+In example above badge events will be sent to `badges` event-bus topic.
+
+Badge Processor
+~~~~~~~~~~~~~~~~
+
+For incoming external events processing configuration points to an importable `Badge Processor`_ class definition.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["processor"] = "credentials.apps.badges.processing.BadgeProcessor"
+
+Badge Collector
+~~~~~~~~~~~~~~~~
+
+For completed badges configuration points to an importable `Badge Collector`_ class definition.
+
+.. code-block:: sh
+
+    BADGES_CONFIG["collector"] = "credentials.apps.badges.collecting.BadgeCollector"
+
+
+Badge templates management
+--------------------------
+
+Badge templates management includes the following life-cycle stages or maintenance activities:
+
+- badge template creation
+- badge template configuration
+- badge template activation (deactivation)
+- badge template editing
+- badge template archiving
+
+.. note::
+    System allows badge templates management via:
+
+    - Credentials admin interface
+
+Creation
+~~~~~~~~
+
+Before being used a badge template must be created.
+Each `badge template`_ has at least its basic properties:
+
+- uuid (unique identifier, auto-generated)
+- name (verbose label - currently, out of localization)
+- icon(image)
```

Comment:
Should `status` be part of these properties -- `inactive`, `active`, and `archived`?